vision/diffusion_mnist/diffusion_plot.jl

```diff
@@ -0,0 +1,118 @@
+include("diffusion_mnist.jl")
+
+using Images
+
+"""
+Helper function yielding the diffusion coefficient from a SDE.
+"""
+diffusion_coeff(t, sigma=convert(eltype(t), 25.0f0)) = sigma .^ t
+
+"""
+Helper function that produces images from a batch of images.
+"""
+function convert_to_image(x, y_size)
+    Gray.(
+        permutedims(
+            vcat(
+                reshape.(
+                    chunk(x |> cpu, y_size), 28, :
+                )...
+            ),
+            (2, 1)
+        )
+    )
+end
+
+"""
+Helper function that generates inputs to a sampler.
+"""
+function setup_sampler(device, num_images=5, num_steps=500, ϵ=1.0f-3)
+    t = ones(Float32, num_images) |> device
+    init_x = (
+        randn(Float32, (28, 28, 1, num_images)) .*
+        expand_dims(marginal_prob_std(t), 3)
+    ) |> device
+    time_steps = LinRange(1.0f0, ϵ, num_steps)
+    Δt = time_steps[1] - time_steps[2]
+    return time_steps, Δt, init_x
+end
+
+"""
+Sample from a diffusion model using the Euler-Maruyama method.
+
+# References
+https://yang-song.github.io/blog/2021/score/#how-to-solve-the-reverse-sde
+"""
+function Euler_Maruyama_sampler(model, init_x, time_steps, Δt)
+    x = mean_x = init_x
+    progress = Progress(length(time_steps))
+    @info "Start Euler-Maruyama Sampling"
+    for time_step in time_steps
+        batch_time_step = fill!(similar(init_x, size(init_x)[end]), 1) .* time_step
+        g = diffusion_coeff(batch_time_step)
+        mean_x = x .+ expand_dims((g .^ 2), 3) .* model(x, batch_time_step) .* Δt
+        x = mean_x + sqrt(Δt) * expand_dims(g, 3) .* randn(Float32, size(x))
+        next!(progress; showvalues=[(:time_step, time_step)])
+    end
+    return mean_x
```

Comment:
```suggestion
    @info "Start Euler-Maruyama Sampling"
    @showprogress for time_step in 1:length(time_steps)
        batch_time_step = fill!(similar(init_x, size(init_x)[end]), 1) .* time_step
        g = diffusion_coeff(batch_time_step)
        mean_x = x .+ expand_dims(g, 3) .^ 2 .* model(x, batch_time_step) .* Δt
        x = mean_x .+ sqrt(Δt) .* expand_dims(g, 3) .* randn(Float32, size(x))
    end
    return mean_x
```
This can be simplified further by using `@showprogress "Start Euler-Maruyama Sampling" for ...`.

Reply:
Had to switch
```julia
 @showprogress for time_step in 1:length(time_steps)
```

to 
```julia
 @showprogress for time_step in time_steps
```
But other than that, this works fine